File: LeetCode_Solutions/2800-minimum-string-length-after-removing-substrings/minimum-string-length-after-removing-substrings.py
class Solution:
    def minLength(self, s: str) -> int:
        st = []  # Use list as a stack
        for char in s:
            if not st:  # If the stack is empty
                st.append(char)
            else:
                if char == 'B':
                    if st[-1] == 'A':  # Check the top of the stack
                        st.pop()  # Pop the top of the stack
                    else:
                        st.append(char)  # Push the current character
                elif char == 'D':
                    if st[-1] == 'C':  # Check the top of the stack
                        st.pop()  # Pop the top of the stack
                    else:
                        st.append(char)  # Push the current character
                else:
                    st.append(char)  # Push the current character
        
        return len(st)  # Return the size of the stack

File: LeetCode_Solutions/2800-minimum-string-length-after-removing-substrings/minimum-string-length-after-removing-substrings.java
class Solution {
    public int minLength(String s) {
        Stack<Character> st=new Stack<>();
        for(int i=0;i<s.length();i++){
            if(st.isEmpty()){
                st.push(s.charAt(i));
            }
            else{
                if(s.charAt(i)=='B'){
                    if(st.peek()=='A'){
                        st.pop();
                    }
                    else{
                        st.push(s.charAt(i));
                    }
                }
                else  if(s.charAt(i)=='D'){
                    if(st.peek()=='C'){
                        st.pop();
                    }
                    else{
                        st.push(s.charAt(i));
                    }
                }
                else{
                    st.push(s.charAt(i));
                }
            }
        }
       return st.size();
    }
}